scripts/aznfs_install.sh

```diff
@@ -75,6 +115,93 @@ use_dnf_or_yum()
     fi
 }
 
+is_new_version_available()
+{
+    local current_version_integer=$(echo $1 | tr -d ".")
+    local latest_version_integer=$(echo $2 | tr -d ".")
+
+    [ $latest_version_integer -gt $current_version_integer ]
+}
+
+# Function to perform AZNFS updates
+perform_aznfs_updates() 
+{
+    # For watchdog, the flag file will always be present; otherwise, the service will be "manual-update"
+    if [ -f /tmp/update_in_progress_from_watchdog.flag ] || [ "$RUN_MODE" == "manual-update" ]; then
+        if [ "$install_cmd" == "apt" ]; then
+            AZNFS_RELEASE="aznfs-${RELEASE_NUMBER}-1"
+            package_name=${AZNFS_RELEASE}_amd64.deb
+        elif [ "$install_cmd" == "zypper" ]; then
+            AZNFS_RELEASE_SUSE="aznfs_sles-${RELEASE_NUMBER}-1"
+            package_name=${AZNFS_RELEASE_SUSE}.x86_64.rpm
+        else
+            AZNFS_RELEASE="aznfs-${RELEASE_NUMBER}-1"
+            package_name=${AZNFS_RELEASE}.x86_64.rpm
+        fi
+        
+        wget "https://github.com/Azure/AZNFS-mount/releases/download/${RELEASE_NUMBER}/${package_name}" -P /tmp
+        if [ "$install_cmd" == "zypper" ]; then
+            $install_cmd install --allow-unsigned-rpm -y "/tmp/${package_name}"
+        else
+            $install_cmd install -y "/tmp/${package_name}"
+        fi
+        install_error=$?
+        rm -f "/tmp/${package_name}"
+
+        if [ "$RUN_MODE" == "auto-update" ] && [ "$install_error" -eq 0 ]; then
+            pecho "AZNFS updates installed. Restarting aznfswatchdog to apply changes!"
+            systemctl daemon-reload
+            systemctl restart aznfswatchdog
+            exit 1 # Nothing in the script will run after this point.
+        fi
+    fi
+}
+
+check_and_perform_update()
```

Comment:
In this function, we are not issuing the update command and the name is conflicting with the other perform_aznfs_update function. We can change the name to just "check_aznfs_update"